app/src/test/kotlin/fr/free/nrw/commons/media/MediaClientTest.kt

```diff
@@ -197,6 +203,38 @@ class MediaClientTest {
         assertEquals(media2.filename, "Test")
     }
 
+    @Test
+    fun getMediaListForUser() {
+        val mockContinuation = mapOf("gcmcontinue" to "test")
+        val imageInfo = ImageInfo()
+
+        val mwQueryPage = mock(MwQueryPage::class.java)
+        whenever(mwQueryPage.title()).thenReturn("Test")
+        whenever(mwQueryPage.imageInfo()).thenReturn(imageInfo)
+
+        val mwQueryResult = mock(MwQueryResult::class.java)
+        whenever(mwQueryResult.pages()).thenReturn(listOf(mwQueryPage))
+
+        val mockResponse = mock(MwQueryResponse::class.java)
+        whenever(mockResponse.query()).thenReturn(mwQueryResult)
+        whenever(mockResponse.continuation()).thenReturn(mockContinuation)
+
+        whenever(
+            mediaInterface!!.getMediaListForUser(
+                ArgumentMatchers.anyString(), ArgumentMatchers.anyInt(),
+                continuationCaptor!!.capture()
+            )
+        )
+            .thenReturn(Observable.just(mockResponse))
+        val media1 = mediaClient!!.getMediaListForUser("Test").blockingGet().get(0)
+        val media2 = mediaClient!!.getMediaListForUser("Test").blockingGet().get(0)
+
+        verify(mediaInterface, times(2))?.getMediaListForUser(
```

Comment:
my intention with the verify was to not use argument matchers and have 2 separate calls that verify exact arguments.

I imagine this test invokes `getMediaListForUser` twice for the purpose of testing the continuationStore ternary? `anyMap()` is hiding this fact.

A more robust - but unnecessary - way to approach this would be if continuationStroe was not a hashMap but an object ContinuationStore we could mock and expect values from and then we would have separate tests for when there is and isn't a match in the continuation store

Reply:
Yeah, I agree. This test and other similar tests in `MediaClientTest` can be improved. Can you create a separate issue for it? I have just mimicked the existing `getMediaListFromCategoryTwice` test(i am the original of that test too).

These tests can probably be improved in different PR.

Reply:
https://github.com/commons-app/apps-android-commons/issues/3749 done